docs/encryption.md

```diff
@@ -19,17 +21,151 @@ edited to turn this feature on:
+secret. The Operator itself neither installs Vault, nor configures it; both
+operations should be done manually, as described in the following subsections.
+
+### Installing Vault
+
+The following steps will deploy Vault on Kubernetes with the [Helm 3 package manager](https://helm.sh/). Other Vault installation methods should also work, so the instruction placed here is not obligatory and is for illustration purposes. Read more about installation in Vault’s [documentation](https://www.vaultproject.io/docs/platform/k8s).
+
+1. Add helm repo and install:
+
+    ```bash
+    $ helm repo add hashicorp https://helm.releases.hashicorp.com
+    "hashicorp" has been added to your repositories
+
+    $ helm install vault hashicorp/vault
+    ```
+
+2. After installation, Vault should be first initialized and then *unsealed*.
+    Initializing Vault is done with the following commands:
+
+    ```bash
+    $ kubectl exec -it pod/vault-service-0 -- vault operator init -key-shares=1 -key-threshold=1 -format=json > /tmp/vault-init
+    $ unsealKey=$(jq -r ".unseal_keys_b64[]" < /tmp/vault-init)
+    ```
+
+    To unseal Vault, execute the following command **for each Pod** of Vault
+    running:
+
+    ```bash
+    $ kubectl exec -it pod/vault-service-0 -- vault operator unseal "$unsealKey"
+    ```
+
+### Configuring Vault
+
+1. First, you should enable secrets within Vault. For this you will need a [Vault token](https://www.vaultproject.io/docs/concepts/tokens).
+    Percona Server for MongoDB can use any regular token which allows all operations
+    inside the secrets mount point. In the following example we are using the
+    *root token* to be sure the permissions requirement is met, but actually
+    there is no need in root permissions. We don’t recommend using the root token
+    on the production system.
+
+    ```bash
+    $ cat /tmp/vault-init | jq -r ".root_token"
+    ```
+
+    The output will be like follows:
+
+    ```text
+    s.VgQvaXl8xGFO1RUxAPbPbsfN
+    ```
+
+    Now login to Vault with this token and enable the “psmdb-secret” secrets path:
+
+    ```bash
+    $ kubectl exec -it vault-service-0 -- /bin/sh
+    $ vault login s.VgQvaXl8xGFO1RUxAPbPbsfN
+    $ vault secrets enable --version=1 -path=psmdb-secret kv
+    ```
+
+    !!! note
+
+        You can also enable audit, which is not mandatory, but useful:
+
+        ```bash
+        $ vault audit enable file file_path=/vault/vault-audit.log
+        ```
+
+2. To enable Vault secret within Kubernetes, create and apply the YAML file,
+    as described further.
+
+    1. To access the Vault server via HTTP, follow the next YAML file example:
+
+        ```yaml
+        apiVersion: v1
+        kind: Secret
+        metadata:
+          name: my-cluster-name-vault
+        type: Opaque
+        stringData:
+          keyring_vault.conf: |-
+             token = s.VgQvaXl8xGFO1RUxAPbPbsfN
+             vault_url = vault-service.vault-service.svc.cluster.local
```

Comment:
this is also wrong.
1. `vault-service` will be just `vault`
2. `vault-service.vault-service.svc.cluster.local` implies that you are installing vault in `vault-service` namespace which is not done in the examples above.

So it should be smth like `vault.default.svc.cluster.local` if the user follows the instruction.